### ISSUE TITLE
Python : Improve Python PAM Auth bypass query

### ISSUE THREAD

**Issue by porcupineyhairs** [github/securitylab#706]
### Query PR

https://github.com/github/codeql/pull/10656

### Language

Python

### CVE(s) ID list

N/A

### Report

This is continuation of my other issue, https://github.com/github/securitylab/issues/561.

The issue was scored a low and received a lower bounty. The low score was a surprise as the query was immediately moved to stable after merge. Now that the results are available on LGTM, I can see there are multiple alerts in a file called `pam.py`. All of these seem to be due to a common vulnerable dependency which was copied across multiple projects. I can see for most alerts, the projects don't seem to actually even invoke a call to the enclosing function. I take it, this was the reason for the low score.

I am now proposing new changes to the query. I now add a taint-tracking configuration to check if the vulnerable `pam_authenticate` is actually invoked. This should lead to a very significant reduction in the number of alerts.

With this, I would request that my earlier bug report be updated to the new severity. Please don't create a new Hackerone report for this change. 

### Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc).

- [ ] Yes
- [ ] No

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Query review**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Final decision**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_

**Comment by jorgectf** [comment]
👋  @porcupimeyhairs

Thanks for the submission! We have reviewed your report and validated your findings. After internally assessing the findings and the query we have determined this query doesn't find any new result, and is therefore not eligible for a reward under the Bug Bounty program.

Even though this issue was assessed as being ineligible, we appreciate your effort improving the code of the original query to meet to our higher quality standards, and would still like to offer you $250.

Best regards and happy hacking!
